apps/openmw/mwrender/animation.cpp

```diff
@@ -1748,21 +1748,31 @@ namespace MWRender
 
         if (alpha != 1.f)
         {
-            osg::StateSet* stateset (new osg::StateSet);
+            // If we have an existing material for alpha transparency, just override alpha level
+            osg::StateSet* stateset = mObjectRoot->getOrCreateStateSet();
+            osg::Material* material = static_cast<osg::Material*>(stateset->getAttribute(osg::StateAttribute::MATERIAL));
+            if (material && mAlpha != 1.f)
```

Comment:
mAlpha is equal to alpha in this scope (line 1747), and alpha is not equal to 1. Does this check actually work as it should?

Reply:
My fault. The outer check was just added later than inner one.